src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TagBuilder.cs

```diff
@@ -117,7 +119,7 @@ private void AppendAttributes(StringBuilder sb)
                     continue;
                 }
 
-                var value = WebUtility.HtmlEncode(attribute.Value);
+                var value = _htmlEncoder.HtmlEncode(attribute.Value);
```

Comment:
Consider calling the overload that takes a TextWriter and refactoring the containing method to use StringWriter instead of StringBuilder.